### PR TITLE
Add custom coding matrix support

### PR DESCRIPTION
**Pull request by vitalif**
Hi. Custom matrix is needed, for example, for various LRC implementations

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -184,3 +185,9 @@ func WithFastOneParityMatrix() Option {
 		o.fastOneParity = true
 	}
 }
+
+func WithCustomMatrix(customMatrix [][]byte) Option {
```
Comment (klauspost) [inline options.go:189]: Please document usage.

**Review by klauspost — CHANGES_REQUESTED** [review]
What is the use case for this, if I may ask?

Please add test.

File: options.go (hunk 1 above)
Comment (vitalif) [reply options.go:195]: Ok, added a comment :-)

**Comment by vitalif** [comment]
I think there are at least 2 use cases:
1) interoperability with other libraries that generate the matrix differently or do other hypothetical nasty things like swapping rows
2) LRC (local reconstruction codes) where some parity rows only span several, but not all, data chunks

I added a test, please take a look :-)

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**